cmd/generate-hashes/generate-hashes.go

```diff
@@ -21,8 +25,12 @@ func main() {
 		fmt.Fprintf(os.Stderr, "Usage: generate-hashes <output_directory>\n")
 		os.Exit(1)
 	}
-	outputDir := os.Args[1]
-	err := os.Mkdir(outputDir, 0755)
+	outputDir, err := filepath.Abs(os.Args[1])
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Invalid input: %s", err)
+		os.Exit(1)
+	}
+	err = os.Mkdir(outputDir, 0755)
```

Comment:
I would leave here mask 0777 as this is making not possible for user to have more permissive access on group and other. It represents permissions "before umask".